onnxruntime/core/providers/cpu/tensor/gelu.cc

```diff
@@ -105,4 +119,160 @@ Status Gelu<T>::Compute(OpKernelContext* context) const {
   return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Unsupported approximation_algorithm: ", approximation_algorithm_);
 }
 
+#if defined(__ARM_NEON) && defined(__ARM_FEATURE_FP16_VECTOR_ARITHMETIC)
+
+void ComputeGeluFp16_NEON(const MLFloat16* input, MLFloat16* output, MLFloat16* temp, int64_t count, const std::string& algo) {
+  const float16_t v_half1 = 0.5f;
+  const float16_t v_one1 = 1.0f;
+  const float16_t v_sqrt1_21 = static_cast<float>(M_SQRT1_2);
+  const float16_t v_B1 = 0.7978845608028654f;
+  const float16_t v_C1 = 0.035677408136300125f;
+  const float16_t c1 = 5.0f;
+  const float16_t c2 = -5.0f;
+  const MLAS_FLOAT16X8 v_half = MlasBroadcastF16Float16x8(v_half1);
+  const MLAS_FLOAT16X8 v_one = MlasBroadcastF16Float16x8(v_one1);
+  const MLAS_FLOAT16X8 v_sqrt1_2 = MlasBroadcastF16Float16x8(v_sqrt1_21);
+  const MLAS_FLOAT16X8 v_B = MlasBroadcastF16Float16x8(v_B1);
+  const MLAS_FLOAT16X8 v_C = MlasBroadcastF16Float16x8(v_C1);
+
+  int64_t i = 0;
+
+  if (algo == "tanh") {
```

Comment:
I think all this logic needs to live in MLAS and not in a CPU EP file. is there a limitation that necessitated doing things this way ?